### PR TITLE
Update `flow-go` to its latest version

### PR DESCRIPTION
**Pull request by m-Peter**
## Description

Update `flow-go` to its latest version and fix all the breaking changes.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels 

## Summary by CodeRabbit

- **Refactor**
	- Transitioned dependency management from Ethereum to Flow blockchain across various components, enhancing compatibility and integration.
- **Chores**
	- Updated import paths and dependencies to align with the new blockchain framework, ensuring system reliability and future maintenance efficiency.
- **Bug Fixes**
	- Corrected variable naming inconsistencies related to blockchain network identifiers, improving code clarity and reducing potential errors.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -205,3 +195,5 @@ require (
 )
 
 replace github.com/onflow/crypto => github.com/onflow/crypto v0.24.9
+
+replace github.com/ethereum/go-etherem => github.com/onflow/go-ethereum v1.13.4
```
Comment (sideninja) [inline go.mod:199]: why is this needed?

File: services/requester/requester.go [hunk 2]
```diff
@@ -468,14 +468,6 @@ func cadenceStringToBytes(value cadence.Value) ([]byte, error) {
 // TODO(m-Peter): Consider moving this to flow-go repository
 func getErrorForCode(errorCode uint64) error {
 	switch evmTypes.ErrorCode(errorCode) {
-	case evmTypes.ValidationErrCodeInvalidBalance:
```
Comment (sideninja) [inline services/requester/requester.go:471]: why was this removed?

**Review by sideninja — APPROVED** [review]

File: services/requester/requester.go (hunk 2 above)
Comment (m-Peter) [reply services/requester/requester.go:471]: These 4 errors got removed from upstream in this commit: https://github.com/onflow/flow-go/commit/53b6c8256b9082958162de419ddd309b6d8fffd7

File: go.mod (hunk 1 above)
Comment (m-Peter) [reply go.mod]: So it turns out that it's not needed. Removed in https://github.com/onflow/flow-evm-gateway/pull/201/commits/f27b2f135b7926b6d2f0c00ade08a9af2e9caa94

**Pull request merged**